### PR TITLE
Fix compatibility with Nette 2.4

### PR DESCRIPTION
**Pull request by enumag**

### CODE REVIEW

**Comment by enumag** [comment]
This one is failing because of Kdyby/Monolog. I don't use that one (yet) but I can send PR there as well.

**Comment by enumag** [comment]
https://github.com/Kdyby/Monolog/pull/18

**Comment by enumag** [comment]
@fprochazka You seem to have a habit of releasing new versions after merging these fixes. So this is an attemt to get a new release of Translation as well. :-P

**Comment by fprochazka** [comment]
@enumag well, if I already have my sleeves rolled up... :)

File: src/Kdyby/Translation/DI/TranslationExtension.php [hunk 1]
```diff
@@ -412,7 +412,7 @@ protected function validateResource($format, $file, $locale, $domain)
 
 		try {
 			$def = $builder->getDefinition($this->loaders[$format]);
-			$refl = Reflection\ClassType::from($def->factory ? $def->factory->entity : $def->class);
+			$refl = Reflection\ClassType::from($def->getFactory() ? $def->getEntity() : $def->getClass());
```
Comment (fprochazka) [inline src/Kdyby/Translation/DI/TranslationExtension.php]: Imho this can be changed to `$def->getEntity() ?: $def->getClass()`

Comment (enumag) [inline src/Kdyby/Translation/DI/TranslationExtension.php]: Right. :+1:

**Comment by fprochazka** [comment]
Thank you!

**Comment by enumag** [comment]
The build didn't pass yet you know... Packagist didn't catch up with Monolog...

**Comment by fprochazka** [comment]
It passed on my localhost :)

**Comment by fprochazka** [comment]
The `--prefer-lowest` might fail, but I'll fix that.

**Comment by enumag** [comment]
Alright then. Thanks. :-) I was trying to test it on my localhost as well but some tests are failing there for CRLF reasons. :-(

**Comment by fprochazka** [comment]
@enumag your wish [has been granted](https://github.com/Kdyby/Translation/releases/tag/v2.3.0) :)

**Comment by enumag** [comment]
@fprochazka Awesome, thank you! :-)

**Pull request closed without merging**